pkg/controller/service/service_controller.go

```diff
@@ -264,7 +264,9 @@ func (s *ServiceController) createLoadBalancerIfNeeded(key string, service *v1.S
 		needDelete := true
 		_, exists, err := s.balancer.GetLoadBalancer(s.clusterName, service)
 		if err != nil {
-			return fmt.Errorf("Error getting LB for service %s: %v", key, err), retryable
+			err = fmt.Errorf("Error getting LB for service %s: %v", key, err)
+			s.eventRecorder.Event(service, v1.EventTypeWarning,"GettingLoadBalancerFailed", err.Error())
```

Comment:
Seems like we already emit event for error returned by createLoadBalancerIfNeeded(): https://github.com/kubernetes/kubernetes/blob/v1.7.3/pkg/controller/service/servicecontroller.go#L230-L242, we may not want to have duplicated events?

The event message "Error creating load balancer" we put in processServiceUpdate() is confusing though, as it could be any error when processing the update (create/update/delete)...

Reply:
is this event for the end-user or the admin?

what is a user of a namespace going to realistically do in response to this event?

Reply:
also when would a failure to get the load balancer occur?  for example, if i exceed my aws API quota, will this happen?  and when it happens will i start spamming my API server?

Reply:
@derekwaynecarr - Indeed, it can spam the API server. on the other hand, having events of high-level prevents us to know what exactly is going on.

Reply:
we now have client side spam protection here: https://github.com/kubernetes/kubernetes/pull/47367